pkgs/development/python-modules/lightgbm/default.nix

```diff
@@ -8,8 +8,14 @@
 , scikit-learn
 , llvmPackages ? null
 , pythonOlder
+, python
+, ocl-icd
+, opencl-headers
+, boost
+, gpuSupport ? true
```

Comment:
```suggestion
```
we want to avoid package variants in python as much as possible

Reply:
Avoiding variants will not be possible in this case.
https://github.com/microsoft/LightGBM/tree/master/python-package#build-gpu-version

The package upstream has many "versions", each of which needs to be built with a different flag. Some of these flags work together, and some don't. For example, if we add cuda support to this package,
1. it will need to be disabled by default because of the unfree license
2. enabling it will conflict with the --gpu flag
As this package gets extended to support more of the available features, users will need a way to select which combination of flags their copy is built with depending on the environment they want to run in.